core/src/gas.rs

```diff
@@ -150,6 +150,18 @@ impl GasCounter {
             burned: self.burned,
         }
     }
+
+    /// Clone the counter
+    ///
+    /// # Safety
+    ///
+    /// Use only when it's absolutely necessary to cloen the counter i.e atomic implementation of `Ext`.
```

Comment:
Same for other copy-pasted clone impls
```suggestion
    /// Clone the counter.
    ///
    /// # Safety
    ///
    /// Use only when it's absolutely necessary to clone the counter i.e atomic implementation of `Ext`.
```